azurelinuxagent/agent.py

```diff
@@ -205,25 +205,36 @@ def collect_logs(self, is_full_mode):
         # Check the cgroups unit
         if CollectLogsHandler.should_validate_cgroups():
 
-            def validate_cgroup_path(path):
+            def validate_cgroup_path(path, expected_slice, expected_unit):
                 if path is None:
                     return False, False
 
-                regex_match = re.match(r'^(?P<slice>.*)/(?P<unit>.*)$', path)
+                # '(.*/)?': process slice may be nested in other, heirarchical slices.
+                # '[^\s./]*': process slice can't contain spaces, periods, or slashes.
```

Comment:
thanks for adding those comments. could you also add the slices we are expecting? i don't quite get why (.*/) is optional (in r'(.*/)?(?P<slice>[^\s./]*.slice)')

Reply:
Maybe the attempt was misguided, but the intent was to make this regex general to any cgroup slice/unit path. I.e. it's possible that our process starts up in `system.slice/tmp.scope` or something, in which case `.*/` (non-optional) would capture `system.slice`, and our regex groups would be off. I think the main benefit would be that we could more easily log expected vs. actual, but the code doesn't actually do that, so...

As for the happy path, I had thought we only  had to worry about `azure.slice/azure-walinuxagent.slice/azure-walinuxagent-periodic_logcollector.slice/collect-logs.scope`  (as that is what I expected from systemd as I understand it), but actually running the code on ubuntus 16, 18, and 20 show this seems to depend w.r.t. systemd version...

So far, I've seen:

ubuntu 20 (systemd v245):
* cpu path: `azure.slice/azure-walinuxagent.slice/azure-walinuxagent-periodic_logcollector.slice` (note the lack of the `*.scope`)
* memory path: `azure.slice/azure-walinuxagent.slice/azure-walinuxagent-periodic_logcollector.slice/collect-logs.scope`

ubuntu 18 (systemd v237, I think):
* cpu path: `azure.slice/azure-walinuxagent.slice/azure-walinuxagent-periodic_logcollector.slice/collect-logs.scope`
* memory path: `azure.slice/azure-walinuxagent.slice/azure-walinuxagent-periodic_logcollector.slice/collect-logs.scope`